tests/system/runtimes/test_kubejob.py

```diff
@@ -716,3 +716,48 @@ def test_job_from_serving_runtime(self):
             v3io_client.object.get("projects", f"{self.project_name}/out.parquet")
         finally:
             v3io_client.close()
+
+    def test_retry_job_exhausted(self):
+        code_path = str(self.assets_path / "raise_func.py")
+
+        function = self.project.set_function(
+            code_path,
+            name="raise-func",
+            kind="job",
+            handler="handler",
+        )
+
+        retry_count = 3
+        retry = mlrun.model.Retry(
+            count=retry_count,
+        )
+
+        with pytest.raises(mlrun.runtimes.utils.RunError):
+            function.run(verbose=True, retry=retry)
+
+        runs = self._run_db.list_runs(project=self.project_name)
+        assert len(runs) == 1
+        run = mlrun.RunObject.from_dict(runs[0])
+        assert run.status.retry_count is None
+        assert (
+            run.status.state == mlrun.common.runtimes.constants.RunStates.pending_retry
+        )
+        assert f"Run failed attempt 1 of {retry_count + 1}" in run.status.status_text
+
```

Comment:
to make the intention clearer
```suggestion
        )
        max_attempts = retry_count + 1
        assert f"Run failed attempt 1 of {max_attempts}" in run.status.status_text

```